### PR TITLE
Change scrollbar behaviour

### PR DESCRIPTION
**Pull request by 0HyperCube**
Change some of the behavour from #328

---
This change is [Reviewable](https://reviewable.io/reviews/graphiteeditor/graphite/351)

### CODE REVIEW

File: editor/src/input/mouse.rs [hunk 1]
```diff
@@ -22,6 +22,10 @@ impl ViewportBounds {
 	pub fn size(&self) -> DVec2 {
 		self.bottom_right - self.top_left
 	}
+
+	pub fn mid(&self) -> DVec2 {
```
Comment (TrueDoctor) [inline editor/src/input/mouse.rs:26]: to stay somewhat consistent with the existing code I would prefer center

File: editor/src/document/document_file.rs [hunk 2]
```diff
@@ -406,15 +406,16 @@ impl MessageHandler<DocumentMessage, &InputPreprocessor> for DocumentMessageHand
...
 				);
-				let root = self.layerdata(&[]);
-				let viewport = ipp.viewport_bounds.size();
-				let [bounds1, bounds2] = self.document.visible_layers_bounding_box().unwrap_or_default();
-				let bounds1 = bounds1.min(DVec2::ZERO) - viewport * (f64::powf(2., root.scale / 3.) * 0.5);
-				let bounds2 = bounds2.max(viewport) + viewport * (f64::powf(2., root.scale / 3.) * 0.5);
+				let scale = 0.5 + 0.5 * self.layerdata(&[]).scale;
+				let viewport_size = ipp.viewport_bounds.size();
+				let viewport_mid = ipp.viewport_bounds.mid();
+				let [bounds1, bounds2] = self.document.visible_layers_bounding_box().unwrap_or([viewport_mid, viewport_mid]);
```
Comment (TrueDoctor) [inline editor/src/document/document_file.rs:412]: could we do `[viewport_mid;2]` ? Afaik DAffine 2 is copy

File: editor/src/input/mouse.rs (hunk 1 above)
Comment (0HyperCube) [reply editor/src/input/mouse.rs:26]: Done

File: editor/src/document/document_file.rs (hunk 2 above)
Comment (0HyperCube) [reply editor/src/document/document_file.rs:412]: Done

File: editor/src/document/document_file.rs [hunk 3]
```diff
@@ -406,7 +406,7 @@ impl MessageHandler<DocumentMessage, &InputPreprocessor> for DocumentMessageHand
 					}
 					.into(),
 				);
-				let scale = 0.5 + 0.5 * self.layerdata(&[]).scale;
+				let scale = 0.7 + 0.3 * self.layerdata(&[]).scale;
```
Comment (Keavon) [inline editor/src/document/document_file.rs:409]: Unfortunately changing 0.5 and 0.5 to 0.7 and 0.3 doesn't seem to really fix the algorithm, which is that we need to add some extra space to the ends at all times. I suggest taking the final calculation, scaling it by 0.9, and adding 0.05. That way you take whatever your calculation was and convert it into something that is 10% smaller and you will therefore always have 5% of the scrollbar track's total width on either end.

Comment (0HyperCube) [reply editor/src/document/document_file.rs]: Done

**Review by Keavon — APPROVED** [review]
Wonderful job! The behavior feels great.

**Pull request merged**